Add models

matifraga opened the pull request:
Add initial version of Particle and Wall models.

matifraga committed:
Add Particle models and Wall representation

matifraga commented:
@juanmbellini No agrege la función que calcula la fuerza porque quizás sea mejor tenerla fuera de la partícula, no lo tenía totalmente definido

juanmbellini commented on src/main/java/ar/edu/itba/ss/hourglass/models/Particle.java:
```diff
@@ -0,0 +1,222 @@
+    //TODO: Don't know if we are going to need this.
+    @Override
+    public void collide(final Particle other) {
+        final Vector2D deltaR = getDeltaR(other);
```
¿Esta formula de donde viene?

juanmbellini commented on src/main/java/ar/edu/itba/ss/hourglass/models/Collisionable.java:
```diff
@@ -0,0 +1,14 @@
+     * @param particle The {@link Particle} colliding.
+     */
+    void collide(final Particle particle);
+}
```
Agregar end of line al final

juanmbellini commented on src/main/java/ar/edu/itba/ss/hourglass/models/Wall.java:
```diff
@@ -0,0 +1,175 @@
+            return finalPoint;
+        }
+    }
+}
```
Idem

juanmbellini commented on src/main/java/ar/edu/itba/ss/hourglass/utils/ParticleProvider.java:
```diff
@@ -0,0 +1,51 @@
+
+    public final Set<Particle> particles;
+
+    private static final double MASS = 0.01;
```
Estas propiedades las pondría en el "archivo de properties" como defaults, y que puedan parametrizarse a gusto.

juanmbellini committed:
rearrange stuff

- Make all methods receive a Vector instead of 'x' and 'y' components
- Improve the doOverlap method (using Vector2D's API)
- Remove the Collisionable interface (not used in this model)
- Add the acceleration field to Particle (there are forces acting in the system)
- Add public setters to Particle (though they might be removed)

juanmbellini committed:
Add missing end of lines
